### ISSUE TITLE
Source-map refresh question

### ISSUE THREAD

**Issue by merqlove** [rails/sprockets#344]
How to setup it right for guaranteed source-map refresh?

Currently i have new branch of Rails 4.2 with Sprockets 4.0 beta2.

TL;DR
Repo with reproduce:
https://github.com/merqlove/sprockets4-source-map-bug

Added manifest file with source like this:

```javascript
// assets/config/manifest.js
//= link_tree ../images
//= link_directory ../javascripts .js
//= link_directory ../stylesheets .scss

//= link application.js
//= link application.css
```

Also i have root `coffeescript` file:

```coffeescript
# assets/javascripts/application.coffee

#= require web/index
```

And web folder index file:

```coffeescript
# assets/javascripts/web/index.coffee
#= require_self
...
#= require_tree ./broadcasts
...
# some code
```

**First time SourceMap processor generates right version of the map.**
But, if i did some changes in the content, for example its child file with `coffeescript` source in `./broadcasts`, content of `application....js` is changing, for example i'm added `console('LALALA')` but SourceMap not refreshing, always same source-map file with same fingerprint.
4 times i've looks to the end of `application....js`, and 4 times i found:
```
//# sourceMappingURL=application.js-64eb1dee0f03a6038c4dd3bd9f77a25f7b5c05095344c19b5800bba7289b443e.map
//# sourceMappingURL=application.js-64eb1dee0f03a6038c4dd3bd9f77a25f7b5c05095344c19b5800bba7289b443e.map
//# sourceMappingURL=application.js-64eb1dee0f03a6038c4dd3bd9f77a25f7b5c05095344c19b5800bba7289b443e.map
//# sourceMappingURL=application.js-64eb1dee0f03a6038c4dd3bd9f77a25f7b5c05095344c19b5800bba7289b443e.map
```

Same picture also i've seen with `master` branch of current repo.

**Comment by merqlove** [comment]
Yes, it is.
I've added `raise "SOMETHING ..."` inside `source_map_processor.rb` and its not failed on changes.
But `coffee_script_processor.rb` is ok, it's failed with the same.

**Comment by merqlove** [comment]
Seems that something with cache.
Because i see map changes in coffee processor.

**Comment by merqlove** [comment]
![Script works well, as it updated, look console.](https://cloud.githubusercontent.com/assets/18354/16966234/a9ecfa6a-4e14-11e6-9e52-2cbe7ff1f88d.png)

![Map is old](https://cloud.githubusercontent.com/assets/18354/16966237/b1649136-4e14-11e6-8bc0-97dcd9a0fd16.png)

So here screenshot and repo with reproduce:
https://github.com/merqlove/sprockets4-source-map-bug

**Comment by Strech** [comment]
I will try to debug and fix it (I hope, that I can find a solution or point to the root cause)

**Comment by merqlove** [comment]
@Strech 
Example, how SourceMap can look out from Webpack:
![screenshot 2016-08-15 19 31 42](https://cloud.githubusercontent.com/assets/18354/17671526/7ebbcfc8-6328-11e6-829b-249c661b194b.png)

**Comment by Strech** [comment]
@merqlove I've succeeded with reproducible failure in tests. Hopefully, I will nail this issue 😲

**Comment by schneems** [comment]
Thanks for this repro case, it made verifying the fix way easier. Thank you all for your detailed bug report and for fix.

**Comment by merqlove** [comment]
Thanks! Seems it works. Tested on staging project.
